pkg/resourcedefinitions/helper.go

```diff
@@ -230,6 +230,10 @@ func alignSchemaRef(nb map[string]any, key string, lsr, rsr *openapi3.SchemaRef)
 		lv.WriteOnly = rv.WriteOnly
 	}
 
+	if !lv.ReadOnly && rv.ReadOnly {
+		lv.ReadOnly = rv.ReadOnly
+	}
+
```

Comment:
Do we need to merge `immutable` attributes as well?

Reply:
https://github.com/OAI/OpenAPI-Specification/issues/2720 looks like there is no `immutable` in OpenAPI at present